sqlccl: deflake TestExplainGist when run with concurrent ALTER PK

spilchen opened the pull request:
TestExplainGist occasionally fails when a query using a secondary index tries to fetch a column not included in that index (see issue #130282). This change doesn’t address the root cause, but instead ignores the error when it occurs. I've also created a more reliable reproducer in the TestDMLInjectionTest, which we can use to validate the eventual fix (#133129).

Epic: none
Closes #130282
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/133130)
<!-- Reviewable:end -->

rafiss commented on pkg/sql/schemachanger/dml_injection_test.go:
```diff
@@ -523,6 +537,12 @@ func TestAlterTableDMLInjection(t *testing.T) {
+								// If a query is provided, run it without checking the results—just
+								// ensure it doesn't fail.
+								if tc.query != "" {
+									_ = sqlDB.QueryStr(t, tc.query)
```
super nit: this could use `Exec` instead of `QueryStr` since we don't care about the results.

spilchen committed:
sqlccl: deflake TestExplainGist when run with concurrent ALTER PK

TestExplainGist occasionally fails when a query using a secondary index
tries to fetch a column not included in that index (see issue #130282).
This change doesn’t address the root cause, but instead ignores the
error when it occurs. I've also created a more reliable reproducer in
the TestDMLInjectionTest, which we can use to validate the eventual fix
(#133129).

Epic: none
Closes #130282
Release note: none

michae2 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/133130#-:-O9u4-ay7ZcKz1SoTcOV:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r=rafiss,michae2

spilchen commented:
@rafiss do you think we should backport this?

rafiss commented:
I believe the test was only flaking on master, so there's not a strong need to backport. However, since the test is test-only, there's very low risk, so we may as well if there are no conflicts that need to be resolved manually.

spilchen commented:
blathers backport 23.1 23.2 24.1 24.2 24.3

spilchen commented:
We’re only doing automatic backports to 24.2 and 24.3, so we’ll stick with those versions.
